CMakeLists.txt

```diff
@@ -1,23 +1,39 @@
-cmake_minimum_required (VERSION 3.10)
-set(CMAKE_CXX_STANDARD 11)
-project (qseed CXX)
+cmake_minimum_required(VERSION 3.10)
+set(CMAKE_CXX_STANDARD 14)
+set(CMAKE_CXX_STANDARD_REQUIRED ON)
+set(CMAKE_CXX_EXTENSIONS OFF)
 
-# build the executable
-add_executable (qseed eaas.cc qseed.cc)
+project(qseed CXX)
+option(ENABLE_TESTS "Generate a test executable" OFF)
 
-# include header files
-target_include_directories(qseed PRIVATE "./inc")
+find_package(CURL REQUIRED)
+find_package(RapidJSON REQUIRED)
+find_library(SOFTHSM2_LIBRARY NAMES softhsm2 REQUIRED HINTS /usr/local/lib/softhsm)
```

Comment:
I dont know how to do this, but I want this to depend on a cryptoki library, not the softhsm library.

Reply:
Andrew - I was also wondering if this is possible & can move SoftHSM to the test module